src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/PipelineTests.cs

```diff
@@ -59,20 +60,21 @@ protected override bool ShouldRetry(HttpPipelineMessage message, int retry, out
 
         class NullPipelineContext : HttpPipelineRequest
         {
-            public override void SetRequestLine(HttpVerb method, Uri uri)
+            public override void AddHeader(HttpHeader header)
             {
             }
 
-            public override void AddHeader(HttpHeader header)
+            public override bool TryGetHeader(string name, out string value)
             {
+                value = null;
+                return false;
             }
 
-            public override void SetContent(HttpPipelineRequestContent content)
+            public override IEnumerable<HttpHeader> GetHeaders()
```

Comment:
Should it be a property called Headers? Why did you make it GetHeaders method?

Reply:
I don't have a strong opinion. My feeling is that if people see headers property they would try to find TryGetHeader/AddHeader/etc methods there (and I think we should put them there).

Reply:
For my two cents, I'm of the same opinion as Pavel.   I think the expectation is that the expectation was set in ASP.NET for having header manipulation methods and that it makes sense to echo that pattern here.